backend-api/app/routers/system.py

```
@@ -321,32 +290,200 @@ async def run_async(cmd: list[str], timeout: int = 6):
+
+    try:
+        # Download
+        dl = subprocess.run(
+            ["wget", "-q", "-O", deb_path, url],
+            capture_output=True, text=True, timeout=120,
+        )
+        if dl.returncode != 0:
+            raise HTTPException(status_code=502, detail=f"Download failed: {dl.stderr[:500]}")
+
+        # Install
+        inst = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)